cmd/list/account/cmd.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package account
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/output"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "account-roles",
+	Aliases: []string{"accountrole", "account-role", "accountroles"},
+	Short:   "List account roles and policies",
+	Long:    "List account roles and policies for the current AWS account.",
+	Example: `  # List all account roles
+  rosa list account-roles`,
+	Args: cobra.NoArgs,
+	Run:  run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.SortFlags = false
+
+	output.AddFlag(Cmd)
+}
+
+func run(_ *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+	accountRoles, err := awsClient.ListAccountRoles()
+
+	if err != nil {
+		reporter.Errorf("Failed to get account roles: %v", err)
+		os.Exit(1)
+	}
+
+	if len(accountRoles) == 0 {
+		reporter.Infof("No account roles available")
+		os.Exit(0)
+	}
+	if output.HasFlag() {
+		err = output.Print(accountRoles)
+		if err != nil {
+			reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+		os.Exit(0)
+	}
+
+	// Create the writer that will be used to print the tabulated results:
+	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
+	fmt.Fprintf(writer, "ROLE NAME\tROLE TYPE\tROLE ARN\tOPENSHIFT VERSION\n")
+	for _, accountRoles := range accountRoles {
+		fmt.Fprintf(
+			writer,
+			"%s\t%s\t%s\t%s\n",
+			accountRoles.RoleName,
+			accountRoles.RoleType,
```

Comment:
I think this should use a map:
- installer: installer
- support: support
- instance_controlplane: control plane
- instance_worker: worker

Alternatively, since we're filtering for roles created following the `rosa` tag structure and naming convention, the names are descriptive enough that we may not need it? What do you think?

Reply:
Do you mean grouping the roles together as installer, support and displaying them as groups

Reply:
no, I mean just replacing the output of the table, so instead of showing `instance_controlplane`, it shows `control plane`
```go
roleTypeMap := map[string]string{
  "installer":             "installer",
  "support":               "support",
  "instance_controlplane": "control plane",
  "instance_worker":       "worker",
}
```
then in the tab print you call `roleTypeMap[accountRoles.RoleType]`

However, since the roles are all created by `rosa`, the naming convention already contains this info. So maybe we don't need the `ROLE TYPE` column... not sure